src/MonoMod.Backports/System/Memory,is_fx,lt_core_2.1,lt_std_2.1/Runtime/InteropServices/MemoryMarshal.Portable.cs

```diff
@@ -72,25 +72,13 @@ public static ReadOnlySpan<byte> AsBytes<T>(ReadOnlySpan<T> span)
         /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to the location where the 0th element
         /// would have been stored. Such a reference can be used for pinning but must never be dereferenced.
         /// </summary>
-        public static ref T GetReference<T>(Span<T> span)
-        {
-            if (span.Pinnable == null)
-                unsafe { return ref Unsafe.AsRef<T>(span.ByteOffset.ToPointer()); }
-            else
-                return ref Unsafe.AddByteOffset<T>(ref span.Pinnable.Data, span.ByteOffset);
-        }
+        public static ref T GetReference<T>(Span<T> span) => ref span.DangerousGetPinnableReference();
 
         /// <summary>
         /// Returns a reference to the 0th element of the ReadOnlySpan. If the Span is empty, returns a reference to the location where the 0th element
         /// would have been stored. Such a reference can be used for pinning but must never be dereferenced.
         /// </summary>
-        public static ref T GetReference<T>(ReadOnlySpan<T> span)
-        {
-            if (span.Pinnable == null)
-                unsafe { return ref Unsafe.AsRef<T>(span.ByteOffset.ToPointer()); }
-            else
-                return ref Unsafe.AddByteOffset<T>(ref span.Pinnable.Data, span.ByteOffset);
-        }
+        public static ref T GetReference<T>(ReadOnlySpan<T> span) => ref Unsafe.AsRef(in span.GetPinnableReference());
```

Comment:
Do we want to make this return `ref readonly` as well? Is that a breaking change worth caring about?

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.memorymarshal.getreference?view=net-8.0#system-runtime-interopservices-memorymarshal-getreference-1(system-readonlyspan((-0)))

Reply:
Fair enough.